### ISSUE TITLE
Warn if composables are used in the wrong places

### ISSUE THREAD

**Issue by mukundshah** [antfu/eslint-config#259]
### Clear and concise description of the problem

Composable, specially those which use `getCurrentInstance` directly or indirectly at any level, should not be used after awaited expression in `async setup()`.

[Further reference here.](https://vuejs.org/guide/reusability/composables.html#usage-restrictions)

### Suggested solution

As, it's an convention [to name composable functions with camelCase names that start with "use"](https://vuejs.org/guide/reusability/composables.html#naming), based on that, we can add a rule to warn or maybe error when in inappropriate order is used, with some exceptions.

This has been discussed before in https://github.com/vuejs/eslint-plugin-vue/issues/2093

As they have commented, there would be a lot of false positives. An alternate solution to that problem would be to have an allow list or pattern.

So, in `eslint-plugin-antfu` we could add a rule like:

```json
"rules": {
  "composable-order": ["warn", {"allow": ["useComposableWhichCanBeCalledAfterAwaitedExpression"]}]
}
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by antfu** [comment]
In short, no.

A few reasons:

- This repo is a config, not a plugin. Even if for `eslint-plugin-antfu`, it should be handled in `eslint-plugin-vue` 
- `getCurrentInstance` is not a public Vue API: https://vuejs.org/api/
- `<script setup>` handles the async problem, which should solve like 90% of the situation.
- `getCurrentInstance` and hooks like `onMounted` throw runtime warning already.
- It's impossible to cover all cases, either a lot of false positives, or false negatives, due to the fact JS is very dynamic.

I think in general, this is something not worth the effort (as the final result won't be perfect anyway). I would think it's better for users to understand better how composition API works, rather than babysitting on this.

**Comment by so1ve** [comment]
`vue/no-restricted-call-after-await` `vue/no-lifecycle-call-after-await` `vue/no-watch-call-after-await`
`vue/no-expose-call-after-await`

Good? :)
